internal/cmd/controller/gitops/reconciler/gitjob_controller.go

```diff
@@ -77,54 +77,54 @@ func (r *GitJobReconciler) SetupWithManager(mgr ctrl.Manager) error {
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.15.0/pkg/reconcile
 func (r *GitJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	logger := log.FromContext(ctx).WithName("gitjob")
-	var gitRepo v1alpha1.GitRepo
+	gitrepo := &v1alpha1.GitRepo{}
```

Comment:
Renaming aside, I liked having this variable defined as a value instead of a pointer, which made eg. `Get` and `updateStatus` calls more explicit about mutating `gitRepo`, since we would reference it.

Reply:
I didn't copy the commit messages into the PR description, but the short form with a pointer is just so much more common in my opinion.
Sure, mutation becomes more obvious, but there are lot's of functions, e.g. `createJobRBAC`, which expect a pointer, but don't mutate. While get and updatestatus always do mutate, as indicated by their name.